File: test/plugin/windows-cpp/TestEvents.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.

#include "pch.h"
#include "TestEvents.h"

using namespace Microsoft::VisualStudio::C3P::Test;

TestEvent::TestEvent(int counter)
{
    this->counter = counter;
}

int TestEvent::Counter::get()
{
    return this->counter;
}

int TestEvents::staticCounter = 0;

TestEvents::TestEvents()
{
}

void TestEvents::RaiseStaticEvent()
{
    TestEvent^ e = ref new TestEvent(++TestEvents::staticCounter);
    TestEvents::StaticEvent(nullptr, e);
}

void TestEvents::RaiseInstanceEvent()
{
    TestEvent^ e = ref new TestEvent(++this->instanceCounter);
    this->InstanceEvent(this, e);
}

File: test/plugin/windows-cs/TestEvents.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.

using System;

namespace Microsoft.C3P.Test
{
    public sealed class TestEvents
    {
        static int _staticEventCounter;
        int _instanceEventCounter;

        public TestEvents()
        {
        }

        public static event EventHandler<TestEvent> StaticEvent;

        public static void RaiseStaticEvent()
        {
            TestEvent e = new TestEvent(++_staticEventCounter);
            EventHandler<TestEvent> handler = TestEvents.StaticEvent;
            if (handler != null)
            {
                handler(null, e);
            }
        }

        public event EventHandler<TestEvent> InstanceEvent;

        public void RaiseInstanceEvent()
        {
            TestEvent e = new TestEvent(++_instanceEventCounter);
            EventHandler<TestEvent> handler = this.InstanceEvent;
            if (handler != null)
            {
                handler(this, e);
            }
        }
    }
}
